monix-execution/shared/src/main/scala_2.12/monix/execution/FutureUtils.scala

```diff
@@ -66,18 +66,24 @@ object FutureUtils {
   def timeoutTo[A](source: Future[A], atMost: FiniteDuration, fallback: => Future[A])
     (implicit s: Scheduler): Future[A] = {
 
-    val promise = Promise[A]()
+    val promise = Promise[Either[Unit, Try[A]]]()
```

Comment:
`Either[Unit, X]` is isomorphic to `Option[X]` for any `X`, and that would save allocations on `Left(())` vs `None`

Reply:
:man_facepalming: 
Fixed